openapi/openapi_spec_resource_schema_definition_property_test.go

```diff
@@ -267,6 +267,242 @@ func TestIsReadOnly(t *testing.T) {
+				So(valueType, ShouldEqual, schema.TypeList)
+			})
+		})
+	})
+}
+
+func TestIsTerraformListOfSimpleValues(t *testing.T) {
+	Convey("Given a swagger schema definition that has a property of type 'list' with elements of type string", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "list_prop",
+			Type:           typeList,
+			ArrayItemsType: typeString,
+		}
+		Convey("When isTerraformListOfSimpleValues method is called", func() {
+			isTerraformListOfSimpleValues, listSchema := s.isTerraformListOfSimpleValues()
+			Convey("Then the result should be true", func() {
+				So(isTerraformListOfSimpleValues, ShouldBeTrue)
+			})
+			Convey("And the returned schema should be of tupe schema.Schema", func() {
+				So(reflect.TypeOf(*listSchema), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+			})
+		})
+	})
+	Convey("Given a swagger schema definition that has a property of type 'list' with elements of type int", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "list_prop",
+			Type:           typeList,
+			ArrayItemsType: typeInt,
+		}
+		Convey("When isTerraformListOfSimpleValues method is called", func() {
+			isTerraformListOfSimpleValues, listSchema := s.isTerraformListOfSimpleValues()
+			Convey("Then the result should be true", func() {
+				So(isTerraformListOfSimpleValues, ShouldBeTrue)
+			})
+			Convey("And the returned schema should be of tupe schema.Schema", func() {
+				So(reflect.TypeOf(*listSchema), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+			})
+		})
+	})
+	Convey("Given a swagger schema definition that has a property of type 'list' with elements of type float", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "list_prop",
+			Type:           typeList,
+			ArrayItemsType: typeFloat,
+		}
+		Convey("When isTerraformListOfSimpleValues method is called", func() {
+			isTerraformListOfSimpleValues, listSchema := s.isTerraformListOfSimpleValues()
+			Convey("Then the result should be true", func() {
+				So(isTerraformListOfSimpleValues, ShouldBeTrue)
+			})
+			Convey("And the returned schema should be of tupe schema.Schema", func() {
+				So(reflect.TypeOf(*listSchema), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+			})
+		})
+	})
+	Convey("Given a swagger schema definition that has a property of type 'list' with elements of type bool", t, func() {
+		s := &specSchemaDefinitionProperty{
+			Name:           "list_prop",
+			Type:           typeList,
+			ArrayItemsType: typeBool,
+		}
+		Convey("When isTerraformListOfSimpleValues method is called", func() {
+			isTerraformListOfSimpleValues, listSchema := s.isTerraformListOfSimpleValues()
+			Convey("Then the result should be true", func() {
+				So(isTerraformListOfSimpleValues, ShouldBeTrue)
+			})
+			Convey("And the returned schema should be of tupe schema.Schema", func() {
+				So(reflect.TypeOf(*listSchema), ShouldEqual, reflect.TypeOf(schema.Schema{}))
+			})
+		})
+	})
```

Comment:
Primitive tests could be written in a loop

Reply:
as discussed, tackling this in a separate changeset